src/validateWebSite.ts

```diff
@@ -24,13 +24,32 @@ type WebError = {
 
 interface IValidateProperties {
     statusCodes?: string; // [[code,elapsedSeconds], [code,elapsedSeconds]...]
+    canceled?: 'true' | 'false';
 }
 
 const initialPollingIntervalMs = 5000;
 const pollingIncrementMs = 0; // Increase in interval each time
-const maximumValidationMs = 60 * 1000; //
+const maximumValidationMs = 60 * 1000;
+
+interface ICancellation {
+    canceled: boolean;
+}
+const cancellations = new Map<string, ICancellation>();
+
+export function cancelWebsiteValidation(siteTreeItem: SiteTreeItem): void {
+    const cancellation = cancellations.get(siteTreeItem.id);
+    if (cancellation) {
+        cancellations.delete(siteTreeItem.id);
+        cancellation.canceled = true;
+    }
+}
 
 export async function validateWebSite(siteTreeItem: SiteTreeItem, outputChannel: OutputChannel, telemetryReporter: TelemetryReporter): Promise<void> {
+    cancelWebsiteValidation(siteTreeItem);
```

Comment:
Do you need to call cancelWebsiteValidation again here? Shouldn't this be taken care of since you call it before deploy?

Reply:
Yes, but that assumes the caller is doing the right thing.